workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/Constraint.cs

```diff
@@ -0,0 +1,388 @@
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct QueryConstraint objects.
+    /// </summary>
+    public static class Constraint
+    {
+        private static List<ComponentInterest.QueryConstraint> EmptyList
```

Comment:
If the intent is that you return an immutable list that nobody can ever mess up, why not return an `IReadOnlyList`?

Reply:
So if I return IReadOnlyList and call `.ToList()` when using them in the And/OrConstraint, under the hood that conversion would create a new empty list (using `_emptyArray`)

Would that be a good solution?